extmod/vfs_posix_file.c

```diff
@@ -273,4 +280,49 @@ const mp_obj_vfs_posix_file_t mp_sys_stdin_obj = {{&mp_type_vfs_posix_textio}, S
 const mp_obj_vfs_posix_file_t mp_sys_stdout_obj = {{&mp_type_vfs_posix_textio}, STDOUT_FILENO};
 const mp_obj_vfs_posix_file_t mp_sys_stderr_obj = {{&mp_type_vfs_posix_textio}, STDERR_FILENO};
 
+#if MICROPY_PY_SYS_STDIO_BUFFER
+STATIC const mp_stream_p_t vfs_posix_buffer_stream_p = {
+    .read = vfs_posix_file_read,
+    .write = vfs_posix_file_write,
+    .ioctl = vfs_posix_file_ioctl,
+    .is_text = false,
+};
+MP_DEFINE_CONST_OBJ_TYPE(
+    mp_type_vfs_posix_buffer_inner,
+    MP_QSTR_TextIOWrapper,
+    MP_TYPE_FLAG_ITER_IS_STREAM,
+    print, vfs_posix_file_print,
+    protocol, &vfs_posix_buffer_stream_p,
+    locals_dict, &vfs_posix_rawfile_locals_dict
+    );
+STATIC mp_obj_vfs_posix_file_t mp_sys_stdin_buffer_obj = {{&mp_type_vfs_posix_buffer_inner}, STDIN_FILENO};
+STATIC mp_obj_vfs_posix_file_t mp_sys_stdout_buffer_obj = {{&mp_type_vfs_posix_buffer_inner}, STDOUT_FILENO};
+
+
+STATIC mp_uint_t stdin_buffer_read(mp_obj_t o_in, void *buf, mp_uint_t size, int *errcode) {
+    return vfs_posix_file_read((mp_obj_t *)&mp_sys_stdin_buffer_obj, buf, size, errcode);
+}
+
+STATIC mp_uint_t stdout_buffer_write(mp_obj_t o_in, const void *buf, mp_uint_t size, int *errcode) {
+    return vfs_posix_file_write((mp_obj_t *)&mp_sys_stdout_buffer_obj, buf, size, errcode);
+}
+
+STATIC const mp_stream_p_t vfs_posix_buffer_io_stream_p = {
+    .read = stdin_buffer_read,
+    .write = stdout_buffer_write,
+    .ioctl = vfs_posix_file_ioctl,
+    .is_text = false,
+};
+MP_DEFINE_CONST_OBJ_TYPE(
+    mp_type_vfs_posix_buffer,
+    MP_QSTR_TextIOWrapper,
```

Comment:
`sys_stdio.mphal.c` uses this name.

```suggestion
    MP_QSTR_FileIO,
```